packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -468,6 +468,7 @@ class ImageStreamCompleterHandle {
 /// configure it with the right [ImageStreamCompleter] when possible.
 abstract class ImageStreamCompleter with Diagnosticable {
   final List<ImageStreamListener> _listeners = <ImageStreamListener>[];
+  final List<ImageStreamListener> _peekListeners = <ImageStreamListener>[];
   ImageInfo? _currentImage;
   FlutterErrorDetails? _currentError;
 
```

Comment:
Github isn't letting me comment on the line below directly, but there is a `hasListeners` getter. we need to document how these _peekListeners are treated with regards to hasListeners.